browser/src/Services/QuickOpen.ts

```diff
@@ -4,102 +4,144 @@
+                return
+            }
+
             const fullPath = path.join(arg.detail, arg.label)
 
             this._seenItems.push(fullPath)
 
             if (!selectedItem.openInSplit) {
-                neovimInstance.command("tabnew! " + fullPath)
+                neovimInstance.command("e! " + fullPath)
             } else {
                 neovimInstance.command("vsp! " + fullPath)
             }
+
+            if (arg.icon === "chain") {
+                neovimInstance.chdir(fullPath)
+            }
+
         })
     }
 
-    public show(): void {
+    public async show() {
+        this._loadedItems.splice(0, this._loadedItems.length - 1)
+
         const config = Config.instance()
         const overriddenCommand = config.getValue("editor.quickOpen.execCommand")
-        const exclude = config.getValue("oni.exclude")
+        // const exclude = config.getValue("oni.exclude")
+
+        //  If in exec directory or home, show bookmarks to change cwd to
+        if (this._isInstallDirectoryOrHome()) {
+            const bookmarks = config.getValue("oni.bookmarks")
+            let icon = "chain"
+            const helpMessage = "No bookmarks yet! Select this to open config! (shows example)"
+            let noPush = false
+
+            // If bookmarks are null show a help message and open config on selection
+            // If we have already pushed, don't push again
+            // Disable lint to prevent duplicate ifs
+            if (bookmarks.length === 0 || (noPush = bookmarks[0] === helpMessage)) { // tslint:disable-line no-conditional-assignment
+                icon = "info"
+                if (!noPush) {
+                    bookmarks.push(helpMessage)
+                }
+            }
+            this._showMenuFromFiles(bookmarks, icon)
+            return
+        }
 
-        UI.Actions.showPopupMenu("quickOpen", [{
-            icon: "refresh fa-spin fa-fw",
-            label: "Loading Files...",
-            detail: "",
-            pinned: false,
-        }])
+        let icon = "file-text-o"
 
         // Overridden strategy
         if (overriddenCommand) {
-            try {
-                // replace placeholder ${search} with "" for initial case
-                const files = execSync(overriddenCommand.replace("${search}", ""), { cwd: process.cwd() })
-                    .toString("utf8")
-                    .split("\n")
-                this._showMenuFromFiles(files)
-                return
-            } catch (e) {
-                Log.warn(`'${overriddenCommand}' returned an error: ${e.message}\nUsing default file list`)
-            }
+            // replace placeholder ${search} with "" for initial case
+            this.loadMenu(overriddenCommand.replace("${search}", ""), icon)
+            return
         }
 
         // Default strategy
-        //  If git repo, use git ls-files
-        //  Otherwise, find all files recursively
-        const openPromise = Git.isGitRepository()
```

Comment:
I like that you consolidated this strategy, to reuse the existing code path (and just have `git` be another search command). Thanks!